cfripper/rules/stack_name_matches_regex.py

```diff
@@ -0,0 +1,40 @@
+from typing import Dict, Optional
+
+from pycfmodel.model.cf_model import CFModel
+
+from cfripper.config.regex import REGEX_ALPHANUMERICAL_OR_HYPHEN
+from cfripper.model.enums import RuleMode, RuleRisk
+from cfripper.model.result import Result
+from cfripper.rules.base_rules import Rule
+
+
+class StackNameMatchesRegexRule(Rule):
+    """
+    Checks that a given stack follows the naming convention given by a regex.
+    """
+
+    RULE_MODE = RuleMode.DEBUG
```

Comment:
We need to explain why this rule is on `DEBUG` by default (demo purposes)

Reply:
Leaving it with the default mode should be ok too

Reply:
As a comment in the code or in the pr?

Reply:
comment in the code should be fine